tracer/src/Datadog.Trace/AppSec/Rasp/RaspModule.cs

```diff
@@ -45,14 +50,40 @@ private static void CheckVulnerability(string address, string valueToCheck)
         }
 
         var arguments = new Dictionary<string, object> { [address] = valueToCheck };
-        RunWaf(arguments, rootSpan);
+        RunWaf(arguments, rootSpan, address);
     }
 
-    private static void RunWaf(Dictionary<string, object> arguments, Span rootSpan)
+    private static void RecordTelemetry(string address, bool isMatch, bool timeOut)
+    {
+        if (!_addressRuleType.TryGetValue(address, out var ruleType))
+        {
+            Log.Warning("RASP: Rule type not found for address {Address}", address);
+            return;
+        }
+
+        TelemetryFactory.Metrics.RecordCountRaspRuleEval(ruleType);
+
+        if (isMatch)
+        {
+            TelemetryFactory.Metrics.RecordCountRaspRuleMatch(ruleType);
+        }
+
+        if (timeOut)
+        {
+            TelemetryFactory.Metrics.RecordCountRaspTimeout(ruleType);
+        }
+    }
+
+    private static void RunWaf(Dictionary<string, object> arguments, Span rootSpan, string address)
     {
         var securityCoordinator = new SecurityCoordinator(Security.Instance, SecurityCoordinator.Context, rootSpan);
         var result = securityCoordinator.RunWaf(arguments, runWithEphemeral: true);
 
+        if (result is not null)
+        {
+            RecordTelemetry(address, result.ReturnCode == Waf.WafReturnCode.Match, result.Timeout);
```

Comment:
I find a number of things about this confusing:
- the method name "RecordTelemetry" but it's only telemetry for RASP, so perhaps the name should be more specific
- give that this method only records rasp telemetry, should it always called from the "RunWaf" method? (I think this method is only used for RASP runs of the WAF, but a better name indicating that would be good)

Reply:
Yes, it's only used for RASP. I will update the naming though. Thanks!